### ISSUE TITLE
es-ES locale intent slot missing the uttered value in the Alexa Simulator

### ISSUE THREAD

**Issue by juanpescador** [alexa/alexa-skills-kit-sdk-for-nodejs#420]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

Not sure if this is the right place to report what seems to be a server-side bug, but I couldn't find anywhere more relevant.

The es-ES locale seems to be slightly broken; when a skill is opened and the user utters/types a word, Alexa is not returning the value of the uttered word.

1. In the Alexa simulator, with locale set to es-ES:
2. Open a skill (type in 'abre `<skill name>`')
3. Type in a word, e.g. 'vacuno'

## Expected Behavior

I expect the `request.intent.slots.Item` object to have a `value` property with the uttered word (e.g. 'vacuno'). Note it is a different `value` property from the one that appears on the object in the `resolutionsPerAuthority` array; that value is the canonical word for a group of synonyms.

Here is an en-GB example that shows what I'm expecting:

	"request": {
		"type": "IntentRequest",
		"requestId": "amzn1.echo-api.request.01d591d5-7b5f-4149-bea3-4ac32969fa07",
		"timestamp": "2018-06-25T21:19:46Z",
		"locale": "en-GB",
		"intent": {
			"name": "FoodTemperatureIntent",
			"confirmationStatus": "NONE",
			"slots": {
				"Item": {
					"name": "Item",
					"value": "beef",
					"resolutions": {
						"resolutionsPerAuthority": [
							{
								"authority": "<authority>.LIST_OF_ITEMS",
								"status": {
									"code": "ER_SUCCESS_MATCH"
								},
								"values": [
									{
										"value": {
											"name": "beef",
											"id": "BEEF"
										}
									}
								]
							}
						]
					},
					"confirmationStatus": "NONE"
				}
			}
		}
	}

## Current Behavior

There is no `value` property on the `request.intent.slots.Item` object with the uttered word. In this case I expected `request.intent.slots.Item.value` to be `vacuno`.

    "request": {
		"type": "IntentRequest",
		"requestId": "amzn1.echo-api.request.3e42d482-5452-4ab0-84f2-79eb33565b52",
		"timestamp": "2018-06-25T20:51:28Z",
		"locale": "es-ES",
		"intent": {
			"name": "FoodTemperatureIntent",
			"confirmationStatus": "NONE",
			"slots": {
				"Item": {
					"name": "Item",
					"resolutions": {
						"resolutionsPerAuthority": [
							{
								"authority": "<authority>.LIST_OF_ITEMS",
								"status": {
									"code": "ER_SUCCESS_MATCH"
								},
								"values": [
									{
										"value": {
											"name": "vacuno",
											"id": "BEEF"
										}
									}
								]
							}
						]
					},
					"confirmationStatus": "NONE"
				}
			}
		}
	}

## Steps to Reproduce (for bugs)
1. Set simulator locale to es-ES.
2. Open a skill
3. Type in/utter a word
4. Look at the JsonInput; the request sent to the Lambda function is missing the 'value` property.

## Your Environment

* ASK SDK for Node.js used: none
* Operating System and version: Win 10 x64

## Node.js and NPM Info
* Node.js version used for development: 8.6.0
* NPM version used for development: 5.3.0

**Comment by juanpescador** [comment]
I did some more testing and the bug still appears in the Spanish locale.

I saved my Spanish interaction model to the English locale and it works as I would expect. When I type "ternera" JsonInput has the correct values:

* `request.intent.slots.Item.value === "ternera"` (the uttered word)
* `request.intent.slots.Item.resolutions.resolutionsPerAuthority[0].values[0].value.name === "vacuno"` (the canonical value of which 'ternera' is a synonym).

I saved my English interaction model to the Spanish locale and the bug is reproduced. When I type "veal", which is a synonym of "beef":

* `request.intent.slots.Item` has no `value` property. I would expect it to exist and be "veal".
* `request.intent.slots.Item.resolutions.resolutionsPerAuthority[0].values[0].value.name === "beef"` (which is what I expect, as veal is a synonym of beef).

**Comment by tianrenz** [comment]
Hi @juanpescador , 

could you share your interaction model here?

**Comment by juanpescador** [comment]
Hi @tianrenz, here's the English (en-GB and en-US) and Spanish (es-ES) interaction models:

## English
```
{
  "languageModel": {
    "types": [
      {
        "name": "LIST_OF_ITEMS",
        "values": [
          {
            "id": "BEEF",
            "name": {
              "value": "beef",
              "synonyms": ["pork", "veal", "lamb"]
            }
          },
          {
            "id": "GROUND_MEATS",
            "name": {
              "value": "ground meats",
              "synonyms": []
            }
          },
          {
            "id": "HAM",
            "name": {
              "value": "ham",
              "synonyms": ["smoked ham"]
            }
          },
          {
            "id": "REHEAT_HAM",
            "name": {
              "value": "reheat ham",
              "synonyms": ["fully cooked ham"]
            }
          },
          {
            "id": "POULTRY",
            "name": {
              "value": "poultry",
              "synonyms": ["chicken", "chicken breast", "chicken leg", "chicken thigh", "chicken wings", "ground poultry"]
            }
          },
          {
            "id": "EGGS",
            "name": {
              "value": "eggs",
              "synonyms": []
            }
          },
          {
            "id": "FISH",
            "name": {
              "value": "fish",
              "synonyms": ["shellfish", "shell fish", "sea food"]
            }
          },
          {
            "id": "LEFTOVERS",
            "name": {
              "value": "leftovers",
              "synonyms": []
            }
          },
          {
            "id": "CASSEROLES",
            "name": {
              "value": "casseroles",
              "synonyms": ["casserole"]
            }
          }
        ]
      }
    ],
    "intents": [
      {
        "name": "AMAZON.CancelIntent",
        "samples": []
      },
      {
        "name": "AMAZON.HelpIntent",
        "samples": []
      },
      {
        "name": "AMAZON.RepeatIntent",
        "samples": []
      },
      {
        "name": "AMAZON.StopIntent",
        "samples": []
      },
      {
        "name": "FoodTemperatureIntent",
        "samples": [
          "{Item}",
          "what's the cooking temperature for {Item}",
          "what the cooking temperature for {Item} is",
          "what the cooking temperature for {Item} should be",
          "what's the temperature for {Item}",
          "what's the safe temperature for {Item}",
          "how hot should {Item} be",
          "how hot {Item} should be",
          "how hot does {Item} have to be"
        ],
        "slots": [
          {
            "name": "Item",
            "type": "LIST_OF_ITEMS"
          }
        ]
      }
    ],
    "invocationName": "cooking temperature"
  }
}
```

## Spanish

```
{
  "languageModel": {
    "types": [
      {
        "name": "LIST_OF_ITEMS",
        "values": [
          {
            "id": "BEEF",
            "name": {
              "value": "vacuno",
              "synonyms": ["cerdo", "ternera", "cordero", "carne de vaca", "res", "carne de res"]
            }
          },
          {
            "id": "GROUND_MEATS",
            "name": {
              "value": "carne picada",
              "synonyms": []
            }
          },
          {
            "id": "HAM",
            "name": {
              "value": "jamón",
              "synonyms": ["jamón ahumado"]
            }
          },
          {
            "id": "REHEAT_HAM",
            "name": {
              "value": "jamón recalentado",
              "synonyms": []
            }
          },
          {
            "id": "POULTRY",
            "name": {
              "value": "pollo",
              "synonyms": ["muslo de pollo", "pechuga de pollo", "pechuga", "alita"]
            }
          },
          {
            "id": "EGGS",
            "name": {
              "value": "huevos",
              "synonyms": ["huevo"]
            }
          },
          {
            "id": "FISH",
            "name": {
              "value": "pescado",
              "synonyms": ["marisco"]
            }
          },
          {
            "id": "LEFTOVERS",
            "name": {
              "value": "sobras",
              "synonyms": []
            }
          },
          {
            "id": "CASSEROLES",
            "name": {
              "value": "cocidos",
              "synonyms": ["cocido", "potaje"]
            }
          }
        ]
      }
    ],
    "intents": [
      {
        "name": "AMAZON.CancelIntent",
        "samples": []
      },
      {
        "name": "AMAZON.HelpIntent",
        "samples": []
      },
      {
        "name": "AMAZON.RepeatIntent",
        "samples": []
      },
      {
        "name": "AMAZON.StopIntent",
        "samples": []
      },
      {
        "name": "FoodTemperatureIntent",
        "samples": [
          "{Item}",
          "A qué temperatura se cocina {Item}",
          "A qué temperatura hay que cocer {Item}",
          "Hasta qué temperatura hay que cocer {Item}",
          "Cuál es la temperatura de cocción para {Item}",
          "Qué temperatura de cocción tiene {Item}",
          "Qué temperatura de cocción necesita {Item}",
          "Qué debería ser la temperatura de cocción para {Item}",
          "Cuál es la temperatura para {Item}",
          "Qué temperatura necesita {Item}",
          "Cuál es la temperatura segura para {Item}",
          "Cuál es la temperatura mínima para {Item}",
          "Cómo de caliente debe estar {Item}",
          "El {Item} a qué temperatura debe estar",
          "La {Item} a qué temperatura debe estar",
          "A cuánto hay que cocer {Item}"
        ],
        "slots": [
          {
            "name": "Item",
            "type": "LIST_OF_ITEMS"
          }
        ]
      }
    ],
    "invocationName": "temperatura de cocción"
  }
}
```

**Comment by juanpescador** [comment]
Hi @tianrenz ,

Any idea if this is in fact an issue, or if I've misconfigured something? Certification of my skill failed due to this issue, but I'm stumped as to what else to test. Also if I should report it elsewhere let me know and sorry for the inconvenience.

**Comment by Dyaneleuterio** [comment]
Hi John. I am so sorry for what had happen. There are no issue. I am a
student and learning things. Maybe I must have clicked something at the
alexa gist hub. So sorry again John.

Sorry for any inconvience in your part.

Sincerely,

Dyan

**Comment by tianrenz** [comment]
Hi @juanpescador , 

Sorry for the late reply. This appears to be a service side bug. We would recommend to post your issue on Alexa Developer Forum (see [here](https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues/426)) to get more prompt response from the community. 

Meanwhile, a couple debugging suggestions: 

1. Try saying a different value that resolves to the same id (beef) and see if problem persists
2. Try invoking the skill from a device and see if problem persists

Regards

**Comment by juanpescador** [comment]
Thanks @tianrenz , I tried your suggestion no. 1 with the same results, but can't do no. 2 AFAIK as it's not launched yet.

I've found [a topic](https://forums.developer.amazon.com/questions/174743/problems-with-ask-with-es-es-locale.html) that was opened on the same day as this issue with similar findings (linked in case anyone stumbles upon this). Hopefully it will get some visibility there.

@Dyaneleuterio don't worry, you didn't do anything. You seem to have subscribed to notifications for this issue, that's why you received the email.
